app/src/main/java/com/capstone/event_finder/network/EventDatabase.java

```diff
@@ -0,0 +1,56 @@
+package com.capstone.event_finder.network;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import androidx.annotation.NonNull;
+import androidx.room.Database;
+import androidx.room.Room;
+import androidx.room.RoomDatabase;
+import androidx.sqlite.db.SupportSQLiteDatabase;
+
+import com.capstone.event_finder.interfaces.EventDao;
+import com.capstone.event_finder.models.Event;
+
+@Database(entities = {Event.class}, version = 1)
+public abstract class EventDatabase extends RoomDatabase {
+    private static volatile EventDatabase instance;
+    private static final RoomDatabase.Callback roomCallback = new RoomDatabase.Callback() {
+        @Override
+        public void onOpen(@NonNull SupportSQLiteDatabase db) {
+            super.onOpen(db);
+            new populateDbAsyncTask(instance);
+        }
+    };
+
+    public static synchronized EventDatabase getInstance(Context context) {
+        if (instance == null) {
+            synchronized (EventDatabase.class) {
+                if (instance == null) {
+                    instance = Room.databaseBuilder(context.getApplicationContext(),
+                                    EventDatabase.class, "event_database")
+                            .fallbackToDestructiveMigration()
+                            .addCallback(roomCallback)
+                            .build();
+                }
+            }
+        }
+        return instance;
+    }
+
+    public abstract EventDao eventDao();
+
+    private static class populateDbAsyncTask extends AsyncTask<Void, Void, Void> {
```

Comment:
Realized I forgot to delete that! - That task was originally created because I originally wanted to add a callback where I could start a background thread task that would delete the old database whenever a new one was created and then populate it. I decided to take a different route in clearing the database (instead of just deleting it) before populating it with new events.